### PR TITLE
Rewrite TypeSet checks with new helpers [Part 4]

### PR DESCRIPTION
**Pull request by appilon**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note

```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: aws/resource_aws_default_security_group_test.go [hunk 1]
```diff
@@ -63,16 +59,13 @@ func TestAccAWSDefaultSecurityGroup_classic(t *testing.T) {
...
-						"aws_default_security_group.web", "ingress.3629188364.to_port", "8000"),
-					resource.TestCheckResourceAttr(
-						"aws_default_security_group.web", "ingress.3629188364.cidr_blocks.#", "1"),
-					resource.TestCheckResourceAttr(
-						"aws_default_security_group.web", "ingress.3629188364.cidr_blocks.0", "10.0.0.0/8"),
+					tfawsresource.TestCheckTypeSetElemNestedAttrs("aws_default_security_group.web", "ingress.*", map[string]string{
+						"protocol":      "tcp",
+						"from_port":     "80",
+						"to_port":       "8000",
+						"cidr_blocks":   "1",
```
Comment (anGie44) [inline aws/resource_aws_default_security_group_test.go:66]: small typo: `cidr_blocks.#`

File: aws/resource_aws_default_security_group_test.go [hunk 2]
```diff
@@ -30,16 +30,13 @@ func TestAccAWSDefaultSecurityGroup_basic(t *testing.T) {
...
```
Comment (anGie44) [inline aws/resource_aws_default_security_group_test.go:37]: small typo: `cidr_blocks.#`

File: aws/resource_aws_db_parameter_group_test.go [hunk 3]
```diff
@@ -449,12 +611,16 @@ func TestAccAWSDBParameterGroup_withApplyMethod(t *testing.T) {
...
-					resource.TestCheckResourceAttr(resourceName, "parameter.2478663599.name", "character_set_client"),
-					resource.TestCheckResourceAttr(resourceName, "parameter.2478663599.value", "utf8"),
-					resource.TestCheckResourceAttr(resourceName, "parameter.2478663599.apply_method", "pending-reboot"),
+					tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "parameter.*", map[string]string{
+						"name":         "character_set_server",
+						"value":        "utf8",
+						"apply_method": "immediate",
+					}),
+					tfawsresource.TestCheckTypeSetElemNestedAttrs(resourceName, "parameter.*", map[string]string{
+						"name":         "character_set_server",
```
Comment (anGie44) [inline aws/resource_aws_db_parameter_group_test.go:620]: small typo: `character_set_client`

File: aws/resource_aws_cognito_user_pool_client_test.go [hunk 4]
```diff
@@ -189,28 +187,28 @@ func TestAccAWSCognitoUserPoolClient_allFieldsUpdatingOneField(t *testing.T) {
...
-					resource.TestCheckResourceAttr(resourceName, "allowed_oauth_scopes.2517049750", "openid"),
-					resource.TestCheckResourceAttr(resourceName, "allowed_oauth_scopes.881205744", "email"),
-					resource.TestCheckResourceAttr(resourceName, "allowed_oauth_scopes.2603607895", "phone"),
-					resource.TestCheckResourceAttr(resourceName, "allowed_oauth_scopes.380129571", "aws.cognito.signin.user.admin"),
-					resource.TestCheckResourceAttr(resourceName, "allowed_oauth_scopes.4080487570", "profile"),
+					tfawsresource.TestCheckTypeSetElemAttr(resourceName, "allowed_oauth_flows.*", "openid"),
+					tfawsresource.TestCheckTypeSetElemAttr(resourceName, "allowed_oauth_flows.*", "email"),
+					tfawsresource.TestCheckTypeSetElemAttr(resourceName, "allowed_oauth_flows.*", "phone"),
+					tfawsresource.TestCheckTypeSetElemAttr(resourceName, "allowed_oauth_flows.*", "aws.cognito.signin.user.admin"),
+					tfawsresource.TestCheckTypeSetElemAttr(resourceName, "allowed_oauth_flows.*", "profile"),
```
Comment (anGie44) [inline aws/resource_aws_cognito_user_pool_client_test.go:208]: small typo: `allowed_oauth_scopes`

File: aws/resource_aws_codebuild_project_test.go [hunk 5]
```diff
@@ -1683,24 +1694,19 @@ func TestAccAWSCodeBuildProject_SecondaryArtifacts_ArtifactIdentifier(t *testing
 	bName := acctest.RandomWithPrefix("tf-acc-test-bucket")
 	resourceName := "aws_codebuild_project.test"
 
-	artifactIdentifier1 := "artifactIdentifier1"
-	artifactIdentifier2 := "artifactIdentifier2"
```
Comment (anGie44) [inline aws/resource_aws_codebuild_project_test.go:1687]: Nit: i think you could keep these here and re-use below in the test steps

File: aws/resource_aws_codebuild_project_test.go [hunk 6]
```diff
@@ -1849,23 +1857,19 @@ func TestAccAWSCodeBuildProject_SecondaryArtifacts_Name(t *testing.T) {
 	bName := acctest.RandomWithPrefix("tf-acc-test-bucket")
 	resourceName := "aws_codebuild_project.test"
 
-	name1 := "name1"
-	name2 := "name2"
```
Comment (anGie44) [inline aws/resource_aws_codebuild_project_test.go:1853]: similar comment as above

File: aws/resource_aws_codebuild_project_test.go [hunk 7]
```diff
@@ -1968,24 +1974,19 @@ func TestAccAWSCodeBuildProject_SecondaryArtifacts_Path(t *testing.T) {
 	rName := acctest.RandomWithPrefix("tf-acc-test")
 	resourceName := "aws_codebuild_project.test"
 
-	path1 := "path1"
-	path2 := "path2"
```
Comment (anGie44) [inline aws/resource_aws_codebuild_project_test.go:1972]: similar comment as above

**Comment by appilon** [comment]
Thanks for those catches @anGie44 ! ready for re-review

**Comment by appilon** [comment]
@anGie44 fixed!

**Pull request merged**